### ISSUE TITLE
[dcmtk] unusable on windows / VS2019|VS2022

### ISSUE THREAD

**Issue by malaterre** [microsoft/vcpkg#31395]
### Is your feature request related to a problem? Please describe.

vcpkg/dcmtk cannot currently be used on windows/VS2019. It currently fails with a cryptic error:

```
  foo.cxx
  D:\a\test-vcpkg-dcmtk\test-vcpkg-dcmtk\build\windows-release\vcpkg_installed\x64-windows\include\dcmtk/config/osconfig.h(1138,1): fatal error C1021: invalid preprocessor command 'errorDCMTK' [D:\a\test-vcpkg-dcmtk\test-vcpkg-dcmtk\build\windows-release\foo.vcxproj]
```

ref:
* https://github.com/malaterre/test-vcpkg-dcmtk/actions/runs/4946191041/jobs/8843930225

### Proposed solution

If I understand correctly upstream documentation:

> To reduce DCMTK's implementation complexity, C++11 shall only be used if your compiler conforms closely to the C++11 Standard. As far as we know, this is currently true for the following compilers:
> 
> GNU C++ Compiler (g++) Version 4.8.1 or later.
> Clang (clang++) Version 3.3 or later.

* https://support.dcmtk.org/docs/cxx11_support.html

the port should not add:

` -DCMAKE_CXX_STANDARD=17`

on MSVC compiler

ref:
* https://github.com/microsoft/vcpkg/blob/master/ports/dcmtk/portfile.cmake#LL32C8-L32C32

**Comment by aspfate** [comment]
This is vs behaviour ,use this flag to fix your problem `/Zc:__cplusplus`

**Comment by JonLiu1993** [comment]
@malaterre, Thanks for the investigation, it looks like this issue has been resolved, can I close the issue?

**Comment by michaelonken** [comment]
FYI: We are about to release DCMTK 3.6.8 during the next days (still in 2023). If would mean double work, maybe you want to wait for this.

**Comment by JonLiu1993** [comment]
Fixed by PR #36124
